### ISSUE TITLE
openapi.yaml specs with default enum values generate models with mypy errors

### ISSUE THREAD

**Issue by scottcarr** [koxudaxi/datamodel-code-generator#245]
**Is your feature request related to a problem? Please describe.**

We use mypy and pydantic, so it would be nice if the generated models always passed mypy.  When a property is an enum with a default value, the generated model has a type error (according to mypy).  For example:

```
# openapi.yaml
openapi: 3.0.0
info:
  title: Sample API
  description: Optional multiline or single-line description in [CommonMark](http://commonmark.org/help/) or HTML.
  version: 0.1.9
servers:
  - url: http://api.example.com/v1
    description: Optional server description, e.g. Main (production) server
  - url: http://staging-api.example.com
    description: Optional server description, e.g. Internal staging server for testing
paths:
  /user/{uid}:
    get:
      summary: Returns a user by ID
      description: Optional extended description in CommonMark or HTML.
      parameters:
        - in: path
          name: uid
          schema:
            $ref: "#/components/schemas/UID"
          required: true
          description: user id
      responses:
        '201':    # status code
          description: The user values
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/User"
        '404':
          description: User not found
components:
  schemas:
    UID:
      type: integer
      minimum: 0
    User:
      type: object
      properties:
        uid:
          $ref: "#/components/schemas/UID"
        favorite_color:
          type: string
          enum: [red, green, blue]
          default: red
      required:
        - uid
```

To create the models, I run:

```
datamodel-codegen --input example/openapi.yaml --output foo.py --field-constraints
```

And these are is my models:

```
class UID(BaseModel):
    __root__: int


class FavoriteColor(Enum):
    red = 'red'
    green = 'green'
    blue = 'blue'


class User(BaseModel):
    uid: UID
    favorite_color: Optional[FavoriteColor] = 'red'
```

Then I run mypy on the files in my project:

```
% mypy foo.py
foo.py:25: error: Incompatible types in assignment (expression has type "str", variable has type "Optional[FavoriteColor]")
Found 1 error in 1 file (checked 1 source file)
```

**Describe the solution you'd like**

If the model had this instead:

```
favorite_color: Optional[FavoriteColor] = FavoriteColor.red
```

that would be ideal.  I don't know if it will be difficult to do.

**Describe alternatives you've considered**

I thought about making a custom template that adds `# type: ignore` after all enums with default values, but I thought there might be a better solution.

**Additional context**

This isn't necessarily a problem with datamodel-codegen so I fully understand if you don't want to implement this feature.  I thought you might know of an easy solution.   Thank you!

**Comment by koxudaxi** [comment]
Thank you for creating this issue.
Interesting.
I think I can implement the change.
I assign this issue to me.

**Comment by koxudaxi** [comment]
@scottcarr 
I'm sorry for the late.
I have released a new version as `0.7.0`
The version has an option `--set-default-enum-member`.
The option replaces the default enum value to the enum member. 

```
$ datamodel-codegen --input openapi.yaml --output models.py --input-file-type=openapi --set-default-enum-member --field-constraints
```

models.py
```python
# generated by datamodel-codegen:
#   filename:  openapi.yaml
#   timestamp: 2021-01-31T17:18:10+00:00

from __future__ import annotations

from enum import Enum
from typing import Optional

from pydantic import BaseModel, Field


class UID(BaseModel):
    __root__: int = Field(..., ge=0.0)


class FavoriteColor(Enum):
    red = 'red'
    green = 'green'
    blue = 'blue'


class User(BaseModel):
    uid: UID
    favorite_color: Optional[FavoriteColor] = FavoriteColor.red

```
